apps/studio/src/server/modules/page/__tests__/page.router.test.ts

```diff
@@ -0,0 +1,1206 @@
+      expect(result).toMatchObject({
+        pageId: expect.any(String),
+      })
+      expect(actual).toMatchObject({
+        ...expectedPageArgs,
+        content: createDefaultPage({ layout: "article" }),
+      })
+    })
+
+    it("should create a new page with default Content layout if layout is not provided", async () => {
+      // Arrange
+      const { site } = await setupSite()
+      const expectedPageArgs = {
+        siteId: site.id,
+        title: "Test Page",
+        permalink: "test-page",
+      }
+
+      // Act
+      const result = await caller.createPage({
+        ...expectedPageArgs,
+      })
+
+      // Assert
+      const actual = await db
+        .selectFrom("Resource")
+        .innerJoin("Blob", "Resource.draftBlobId", "Blob.id")
+        .where("Resource.id", "=", result.pageId)
+        .select(["title", "permalink", "type", "siteId", "Blob.content"])
+        .executeTakeFirstOrThrow()
+      expect(result).toMatchObject({
+        pageId: expect.any(String),
+      })
+      expect(actual).toMatchObject({
+        ...expectedPageArgs,
+        content: createDefaultPage({ layout: "content" }),
+      })
+    })
+
+    it("should create a page in folder successfully", async () => {
+      // Arrange
+      const { site, folder } = await setupFolder()
+      const expectedPageArgs = {
+        siteId: site.id,
+        title: "Test Page",
+        permalink: "test-page",
+      }
+
+      // Act
+      const result = await caller.createPage({
+        ...expectedPageArgs,
+        layout: "content",
+        folderId: Number(folder.id),
+      })
+
+      // Assert
+      const actual = await db
+        .selectFrom("Resource")
+        .innerJoin("Blob", "Resource.draftBlobId", "Blob.id")
+        .where("Resource.id", "=", result.pageId)
+        .select([
+          "title",
+          "permalink",
+          "type",
+          "siteId",
+          "Blob.content",
+          "parentId",
+        ])
+        .executeTakeFirstOrThrow()
+      expect(result).toMatchObject({
+        pageId: expect.any(String),
+      })
+      expect(actual).toMatchObject({
+        ...expectedPageArgs,
+        parentId: folder.id,
+        content: createDefaultPage({ layout: "content" }),
+      })
+    })
+
+    it("should throw 400 if folderId does not exist", async () => {
+      // Arrange
+      const { site } = await setupSite()
+      const expectedPageArgs = {
+        siteId: site.id,
+        title: "Test Page",
+        permalink: "test-page",
+        folderId: 999999, // should not exist
+      }
+
+      // Act
+      const result = caller.createPage({
+        ...expectedPageArgs,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "BAD_REQUEST",
+          message: "Folder not found or folderId is not a folder",
+        }),
+      )
+    })
+
+    it("should throw 400 if folderId is not a Folder resource", async () => {
```

Comment:
actually our invariant is that pages can only be created if your parent is `null` or a folder, so we need to check that `folderId` is not a collection too